ddtrace/tracer/option.go

```diff
@@ -325,23 +339,23 @@ func newConfig(opts ...StartOption) *config {
 			return r == ',' || r == ' '
 		})...)(c)
 	}
-	if v := os.Getenv("DD_SERVICE"); v != "" {
+	if v := getDDorOtelConfig("service"); v != "" {
```

Comment:
For the purpose of this PR, it's not needed to implement my following idea: I think at some point we want to encapsulate these configurations with some kind of API that our internal dev users only need to call something like `config.service.get()` and it's automatically resolved.

We have at least four possible sources of configuration (default value, user provided value, environmental variables, dynamic configuration), and it's getting more complex to track down where a config may be set from.

Long term, I envision even some kind of automatic documentation so we can generate from our code and publish in our website.

Reply:
That's a great idea. We do something similar in the python tracer 😄